src/main/java/picard/analysis/CollectQualityYieldMetricsSNVQ.java

```diff
@@ -0,0 +1,550 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.analysis;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.PicardException;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+import picard.util.SeriesStats;
+import picard.util.help.HelpConstants;
+
+import java.io.File;
+import java.util.Vector;
+
+/**
+ * Command line program to calculate SNV quality yield metrics for read files
+ *
+ * @author Dror Kessler
+ */
+
+@CommandLineProgramProperties(
+        summary = CollectQualityYieldMetricsSNVQ.USAGE_SUMMARY + CollectQualityYieldMetricsSNVQ.USAGE_DETAILS,
+        oneLineSummary = CollectQualityYieldMetricsSNVQ.USAGE_SUMMARY,
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+@DocumentedFeature
+public class CollectQualityYieldMetricsSNVQ extends SinglePassSamProgram {
+    private QualityYieldMetricsCollector collector = null;
+    public Histogram<Integer> qualityHistogram = new Histogram<>("KEY", "BQ_COUNT");
+    private Vector<SeriesStats> readPositionQualityStats = new Vector<>();
+    public Histogram<Integer> snvqHistogram = new Histogram<>("KEY", "SNVQ_COUNT");
+    private Vector<SeriesStats> readPositionSnvqStats = new Vector<>();
+    final byte[] baseOrder = "ACGT".getBytes();
```

Comment:
I could not find a constant to use for the list of base values to be used for SNVQ calculation. Converted to a constant - so at least it would not be a literal.